torch/distributed/distributed_c10d.py

```diff
@@ -1125,10 +1125,7 @@ def init_process_group(
         # these barriers may be unnecessary, as proved by a green CI after
         # removal. An environment variable `TORCH_DIST_INIT_BARRIER` has been
         # added which, when set to 0, will disable these barriers.
-        if backend == "fake":
-            # Fake process group doesn't need barrier
-            pass
-        elif backend == Backend.MPI:
+        if backend == Backend.MPI:
```

Comment:
Nyehh, I think I preferred not mucking about with the environment variable instead haha.

Reply:
Yeah I kinda feel bad for this too, it was easier for me to leverage the previous work that uses a environ knob to control barrier init, let me think about it, there should be some better way for custom backend.

Reply:
landing this first to unblock dev efficiency and will address this in a follow up PR

Reply:
follow up PR here: https://github.com/pytorch/pytorch/pull/102343